### PR TITLE
[bugfix]: use tsx instead of ts-node, update @electron/rebuild

### PR DESCRIPTION
**Pull request by kgarner7**
Replace ts-node with tsx for transpile, and upgrade `@electron/rebuild`. It _might_ be possible to fix ts-node to work, but I did not have any luck with that. This requires node >= 18 now

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -295,6 +293,7 @@
         "@ag-grid-community/infinite-row-model": "^28.2.1",
         "@ag-grid-community/react": "^28.2.1",
         "@ag-grid-community/styles": "^28.2.1",
+        "@electron/rebuild": "^3.6.0",
```
Comment (sel10ut) [inline package.json]: Shouldn't it be in `devDependencies`?

File: package.json [hunk 2]
```diff
@@ -350,14 +349,15 @@
         "semver": "^7.5.4",
         "styled-components": "^6.0.8",
         "swiper": "^9.3.1",
+        "tsx": "^4.16.2",
```
Comment (sel10ut) [inline package.json]: and this

**Comment by jeffvli** [comment]
I might just hold off on the v31 upgrade for now. I can't quite get the ESM build to work on [my branch](https://github.com/jeffvli/feishin/tree/fix/esm) either. I got most of it working except for preload which is still breaking.

**Comment by sel10ut** [comment]
it works for me on development with tsx

**Comment by sel10ut** [comment]
try keeping `electron -r ts-node/register/transpile-only` with tsx

**Comment by jeffvli** [comment]
Yep.. working fine for me now. I think when I initially tested, it broke because ts-node wasn't installed. I tried on my own branch and started messing with some ESM stuff but this is much easier with just tsx.

**Pull request merged**